From 32e6469e95ebbb3f8178eb79db3b1d53a2584f91 Mon Sep 17 00:00:00 2001
From: Sergey Larin <cerg2010cerg2010@mail.ru>
Date: Sun, 24 Feb 2019 18:20:08 +0300
Subject: [PATCH] ARM: dts: tegra20-glide: Add bindings for microphone

Mic requires additional GPIOs to work:
- MICBIAS1: internal MICBIAS
- MICBIAS2: external (headphone) MICBIAS
- EAR_SEL: to choose between internal and external mic

Also GPIO 1 on Wolfson chip needs to be asserted.
Make it on by default.

Signed-off-by: Sergey Larin <cerg2010cerg2010@mail.ru>
---
 arch/arm/boot/dts/tegra20-glide.dts | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/arch/arm/boot/dts/tegra20-glide.dts b/arch/arm/boot/dts/tegra20-glide.dts
index d7dfa895cf56..195ccb0a1012 100644
--- a/arch/arm/boot/dts/tegra20-glide.dts
+++ b/arch/arm/boot/dts/tegra20-glide.dts
@@ -868,6 +868,7 @@
 			#gpio-cells = <2>;
 
 			wlf,ldo1ena =  <&gpio TEGRA_GPIO(X, 5) GPIO_ACTIVE_HIGH>;
+			wlf,gpio-cfg = <0xA101 0 0 0 0 0 0 0 0 0 0>;
 
 		};
 	};
@@ -1129,6 +1130,24 @@
 			linux,default-trigger = "gpio";
 			default-state = "off";
 		};
+
+		micbias1 {
+			label = "MICBIAS1";
+			gpios = <&gpio TEGRA_GPIO(X, 6) GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+
+		micbias2 {
+			label = "MICBIAS2";
+			gpios = <&gpio TEGRA_GPIO(H, 3) GPIO_ACTIVE_HIGH>;
+			default-state = "off";
+		};
+
+		earsel {
+			label = "EAR_SEL";
+			gpios = <&gpio TEGRA_GPIO(I, 4) GPIO_ACTIVE_HIGH>;
+			default-state = "off";
+		};
 	};
 
 	pmc@7000e400 {
-- 
2.22.0

